lib/ansible_tower_client/job_template.rb

```diff
@@ -10,12 +10,19 @@ def initialize(api, raw_body)
       super
     end
 
-    def launch(vars = {})
+    def launch(options = {})
       launch_url = "#{url}launch/"
-      extra = JSONValues.new(vars).extra_vars
-      resp = api.post(launch_url, extra).body
-      job = JSON.parse(resp)
-      api.jobs.find(job['job'])
+      new_limit = options.delete('limit')
+      resp = with_temporary_changes(new_limit) do
+        api.post(launch_url, options).body
+      end
+
+      begin
+        job = JSON.parse(resp)
+        api.jobs.find(job['job'])
+      rescue ResourceNotFound
```

Comment:
@bdunne - We did hit this in the past - I was trying to handle it better with the `Error.parse` method. In the past Faraday would raise ResourceNotFound - but the actual text of the that error - e.g. the API response was lost.  If I'm reading what you and @bzwei want - we still need to raise an error - but pass the API response as text shown in the error class message?